src/main/java/org/broadinstitute/hellbender/tools/exome/gcbias/GCCorrector.java

```diff
@@ -0,0 +1,107 @@
+        return IntStream.range(0, NUMBER_OF_GC_BINS).mapToDouble(bin -> {
+            final RealVector weights = new ArrayRealVector(IntStream.range(0, NUMBER_OF_GC_BINS)
+                    .mapToDouble(n -> coveragesByGC.get(n).size() * Math.exp(-Math.abs(bin - n) * correlationDecayRatePerBin)).toArray());
+            return weights.dotProduct(medians) / weights.getL1Norm();
+        }).map(x -> 1/x).toArray();
+    }
+
+    /**
+     *
+     * @param inputCounts raw coverage before GC correction
+     * @param gcContentByTarget      array of gc contents, one per target of the input
+     * @return              GC-corrected coverage
+     */
+    public static ReadCountCollection correctCoverage(final ReadCountCollection inputCounts, final double[] gcContentByTarget) {
+        // each column (sample) has its own GC bias curve, hence its own GC corrector
+        final List<GCCorrector> gcCorrectors = IntStream.range(0, inputCounts.columnNames().size())
+                .mapToObj(n -> new GCCorrector(gcContentByTarget, inputCounts.counts().getColumnVector(n))).collect(Collectors.toList());
+
+        // gc correct a copy of the input counts in-place
+        final RealMatrix correctedCounts = inputCounts.counts().copy();
+        correctedCounts.walkInOptimizedOrder(new DefaultRealMatrixChangingVisitor() {
+            @Override
+            public double visit(int target, int column, double coverage) {
+                return gcCorrectors.get(column).correctedCoverage(coverage, gcContentByTarget[target]);
+            }
+        });
+
+        // we would like the average correction factor to be 1.0 in the sense that average coverage before and after
+        // correction should be equal
+        final double[] columnNormalizationFactors = IntStream.range(0, inputCounts.columnNames().size())
+                .mapToDouble(c -> inputCounts.counts().getColumnVector(c).getL1Norm() / correctedCounts.getColumnVector(c).getL1Norm()).toArray();
+        correctedCounts.walkInOptimizedOrder(new DefaultRealMatrixChangingVisitor() {
+            @Override
+            public double visit(int target, int column, double coverage) {
+                return coverage * columnNormalizationFactors[column];
+            }
+        });
+
+        return new ReadCountCollection(inputCounts.targets(), inputCounts.columnNames(), correctedCounts);
+    }
+
+    private double correctedCoverage(final double coverage, final double gcContent) {
+        return gcCorrectionFactors[gcContentToBinIndex(gcContent)] * coverage;
+    }
+
+    // return a median of coverages or 1 (no correction) if no coverage exists at this gc bin
+    // note that this default value is never used because if the bin is empty, there is nothing to correct
+    private static double medianOrDefault(final List<Double> list) {
+        return list.size() > 0 ? new Median().evaluate(list.stream().mapToDouble(d->d).toArray()) : 1.0;
```

Comment:
1. Magic number.
2. Wouldn't this method be easier to read/use if the List<List<Double>> was turned into a matrix or double[][]?
3.  If you make default a parameter, this would make a reasonable utility.